QtCollider/QcApplication.cpp

```diff
@@ -185,8 +197,12 @@ bool QcApplication::notify(QObject* object, QEvent* event) {
     // This is a hack; for a not-fully-understood reason Qt past 5.7 sends these events to the
     // native window if they aren't accepted here. This caused issue #4058. Accepting them here
     // seems to solve the problem, but might cause other issues since it is a heavy-handed way
-    // of doing this. TODO - solve more elegantly
-    if (result)
+    // of doing this.
+    // In order to still allow closing GUI windows with "cmd-w", we do not accept modifier keys
+    // alone, as well as the "cmd-w" combination itself, so that they propagate further
+    // (see isKeyEvent and isCloseEvent above).
```

Comment:
from this comment, i still don't understand why you filter out modifier keys alone.